src/routes/PaymentNoticePage.tsx

```diff
@@ -47,7 +47,7 @@ export default function PaymentNoticePage() {
 
   const onSubmit = React.useCallback(
     async (notice: PaymentFormFields) => {
-      const rptId: RptId = `${notice.cf}${notice.billCode}`;
+      const rptId: RptId = `${notice.cf}${notice.billCode}` as RptId;
```

Comment:
General comment: it would be better to use proper conversions via `io-ts` instead of type assertions (in this case `RptId.decode`

Reply:
something like that, @giovanniberti right?
```
      const rptId: RptId = pipe(
        RptId.decode(JSON.parse(`${notice.cf}${notice.billCode}`)),
        E.fold((_) => onError(ErrorsType.GENERIC_ERROR), identify)
      );
```

Reply:
In general, yes! This specific snippet would not typecheck as `onError` returns `void` 🙂, but the general idea is there

Reply:
> In general, yes! This specific snippet would not typecheck as `onError` returns `void` 🙂, but the general idea is there

I dunno about this. if you gonna typecheck on runtime the rptid then you should typecheck every other form in the app. On the other hand the form is already checking for correct string formats, so it's impossible to have errors, the E.fold will be always unreachable. If for some reason it's executed it goes against the provided design which expects the FORM to provide the user with useful information about acceptable string formats, and avoid wrong input. 
So the only point of the type assertion here was to be sure that whenever the code is edited in the future, all the types can overlap and not to check if they do on runtime.